crates/wasi-provider/src/states/initializing.rs

```diff
@@ -87,7 +84,10 @@ impl State<PodState> for Initializing {
             // Each new init container resets the CrashLoopBackoff timer.
             pod_state.crash_loop_backoff_strategy.reset();
 
-            let handle = start_container(pod_state, pod, &init_container).await?;
+            let handle = match start_container(pod_state, pod, &init_container).await {
+                Ok(h) => h,
+                Err(e) => fail_fatal!(e),
```

Comment:
Same note here as with the wascc provider. I am not sure if this is a fatal failure, especially from a k8s perspective. Even if a docker container is totally misconfigured, it keeps looping over it again after backoff

Reply:
I still think this should just be an error transition. I think it doesn't follow the k8s patter to have the pod stop executing entirely if there was a flake with something like file or network access during an init pod.

Reply:
@thomastaylor312 I am okay with that, but I want to put it _out of scope for the PR at hand_.  This is a refactoring PR that aims not to (significantly) change observable behaviour.  Changing this to transition to the error state means that pods with failing init containers go into a state where the init container reports itself as terminated but the pod as a whole goes into CrashLoopBackoff.  This currently causes the integration tests to go into an infinite loop waiting for it, and I'm not sure why (possibly it passes the pod ready within 30 seconds test but there isn't a suitable timeout on the completion check).  Obviously I could also fix up the integration tests but it all gets into questions about the correct behaviour and how we test retries and stuff.

So don't get me wrong.  We **should** make this an error transition.  But I feel strongly that we should do so in a feature PR where we have the leisure to discuss the desired behaviour, its ramifications, how we'll test it, and where else we should be making the corresponding changes.  I worry that such a discussion could bog this PR down and I don't want this one to become long-running because I don't want new PRs to have to base themselves on the fallible model.

Reply:
That is fair